social_core/utils.py

```diff
@@ -112,7 +112,7 @@ def sanitize_redirect(hosts, redirect_to):
 def user_is_authenticated(user):
     if user and hasattr(user, 'is_authenticated'):
         if isinstance(user.is_authenticated, collections.Callable):
-            authenticated = user.is_authenticated()
```

Comment:
This is the `core` library, and as such, it expects the `is_authenticated` to exist, but takes into consideration the option of it being a callable, that's the reason for the `if` before it.

Reply:
@omab are you saying that changes in this code block should be reverted ? That is how I understand it, although I don't understand when `is_authenticated` can be a callable. Maybe that applies to other frameworks, and not Django. 

If this is the case then this entire PR should not be needed. Instead just make sure tests cover the 2 branches of the if statement.

Reply:
@atodorov, that's what I'm saying, not reverted, just that this PR is not needed, `is_authenticated()` was deprecated on Django, but this is the core library and needs to take into consideration other frameworks or custom implementations, on Flask is really easy to define your own user model. Ideally, this code should be in the corresponding strategies, have a `strategy.is_authenticated(user)` there that wraps the framework implementation.